shared/js/l10n.js

```diff
@@ -1607,20 +1626,32 @@
       }
     }
 
-    if (!this.ctx.availableLocales.length) {
-      // if there was no availableLanguages meta,
-      // register the default locale only
-      this.ctx.registerLocales(this.ctx.defaultLocale);
+    if (navigator.mozApps && navigator.mozApps.getAdditionalLanguages) {
+      navigator.mozApps.getAdditionalLanguages().then(
+        registerLocales.bind(this));
+    } else {
+      registerLocales.call(this);
     }
     return initLocale.call(this);
   }
 
+  function registerLocales(extraLangs) {
```

Comment:
So, I'm not against it in the current incarnation, but I'd like to make langpacks a bindings feature that library is unaware of. I'd love it to be that only bindings and IO know about different sources of data. 

Do you agree?